### DISCUSSION TITLE
Fixing and sorting our Code

### DISCUSSION THREAD

**Discussion by argastle** [Freaky2164/TowerDefense#23] [General]
Hi,

in this week, we organized our branch-structure and improved our code. We also redesigned our map to a more modern style and adapted our waypoints.
As a new feature we implemented a functional money label, which shows the remaining money. If you place a tower, it gets reduced, if you kill an enemy you get money.
In order to kill enemies you have to start a new round, which is realized by the black button on the right hand corner.

![Screenshot 2022-11-30 193926](https://user-images.githubusercontent.com/45245745/204881419-cd2b4cf9-2aac-4906-92a4-42c3dd126a6c.png)

In this week, we also thought about what software architecture to use. Since we are developing a game and games heavily depend on user interaction, we naturally use the event-driven architecture. It also makes more sense for our project to use the mediator topology, since we plan to implement a central component that handles the different events of our game.

Have a look a look at our utility tree, the performance tactics we use and our design checklist:

![Utility Tree](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Diagrams/Utility%20Tree.png)

![Performance tactics](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Diagrams/Performance%20Tactics%20Highlights.png)

![Design Checklist](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Software%20Architecture%20Documentation/Design%20Checklist%20for%20Performance.md)

Our tasks this week:

Nico: New Map Design / Testing
Paul: Start Round Button /  Money Label
Lukas: Fix Layers / Correct Place Towers
Tim: Fixed the Towers
Johannes: Player Health

Thanks for your contribution!

Tower Defense Team

**Comment by Mohmad-Naser-alnakeshbandi** [comment]
Hey Tower Defense Team,

nice changes 🚀 you rock it as always. 
I have an idea, why not to put the health score in the blue section, so you could have some kind of animation, by losing the score?
If you would like to have this feature in the same position as now, the user has to deal with a lot of animation /movement in the same area (battle area ), and then another section (the blue one) with almost no animation 

Best regards 
Mohamad Naser Alnakeshband

**Comment by MikoHeuken** [comment]
Hello Team Tower Defense,

I like the design of the backround, it looks a little bit more realistic.
Will the towers have diffrent levels. Maybe you can upgrade them with money or if they killed  a few enemys.
One thing that´s disturbing me a little (nothing important) is the backround color on the right. I think blue doesn´t fit with the beautiful backround of the battle. 

Keep going!
Team Recipe Hunter

**Comment by Tinf21-B4-SE-Project** [comment]
Hello Team Tower Defense,

we like your new map design. Will there also now be new tower and enemy designs as they don't match the new map. 
I personally think it would be nice if the text "money" and "health" were replaced by icons and the towers for which you currently have too little money are grayed out.  

Team cybuy
